src/python/sdk/econia_sdk/view/market.py

```diff
@@ -320,6 +320,178 @@ def get_price_levels_all(view: EconiaViewer, market_id: int) -> dict:
+        "get_open_orders_paginated",
+        [],
+        [
+            str(market_id),
+            str(ask_page_size),
+            str(bid_page_size),
+            str(next_ask_idx),
+            str(next_bid_idx),
+        ],
+        ledger_version=ledger_version,
+    )
+    return returns
+
+
+def get_open_orders_with_pagination(
+    viewer: EconiaViewer,
+    market_id: int,
+    max_asks: int = 100,
+    max_bids: int = 100,
+    page_size: int = 100,
+) -> dict:
+    info = viewer.aptos_client.info()
+    ledger_version = int(info["ledger_version"])
+
+    next_ask_idx = "0"
+    next_bid_idx = "0"
+    bid_page_size = ask_page_size = page_size
+
+    asks = []
+    bids = []
+
+    keep_looping = True
+    while keep_looping:
+        returns = get_open_orders_paginated(
+            viewer,
+            market_id,
+            min(ask_page_size, max(max_asks - len(asks), 0)),
+            min(bid_page_size, max(max_bids - len(bids), 0)),
+            int(next_ask_idx),
+            int(next_bid_idx),
+            ledger_version,
+        )
+
+        value = returns[0]
+        next_ask_idx = returns[1]
+        next_bid_idx = returns[2]
+
+        for ask in value["asks"]:
+            asks.append(_convert_open_order_value(ask))
+        for bid in value["bids"]:
+            bids.append(_convert_open_order_value(bid))
+
+        if (len(asks) >= max_asks or next_ask_idx == "0") and (
+            len(bids) >= max_bids or next_bid_idx == "0"
+        ):
+            keep_looping = False
+        elif next_ask_idx == "0" or next_bid_idx == "0":
+            if next_ask_idx == "0":
+                ask_page_size = 0
+            else:
+                bid_page_size = 0
+
+    return {"asks": asks, "bids": bids, "market_id": market_id}
+
+
+def get_price_levels_paginated(
+    viewer: EconiaViewer,
+    market_id: int,
+    ask_page_size: int,
+    bid_page_size: int,
+    next_ask_idx: int,
+    next_bid_idx: int,
+    ledger_version: int,
+):
+    returns = viewer.get_returns(
+        "market",
+        "get_price_levels_paginated",
+        [],
+        [
+            str(market_id),
+            str(ask_page_size),
+            str(bid_page_size),
+            str(next_ask_idx),
+            str(next_bid_idx),
+        ],
+        ledger_version=ledger_version,
+    )
+    return returns
+
+
+def get_price_levels_with_pagination(
+    viewer: EconiaViewer,
+    market_id: int,
+    max_asks: int = 100,
+    max_bids: int = 100,
+    page_size: int = 100,
+) -> dict:
+    info = viewer.aptos_client.info()
+    ledger_version = int(info["ledger_version"])
+
+    next_ask_idx = "0"
+    next_bid_idx = "0"
+    bid_page_size = ask_page_size = page_size
+
+    asks = dict()
+    asks_size = 0
+    bids = dict()
+    bids_size = 0
+
+    keep_looping = True
+    while keep_looping:
+        returns = get_price_levels_paginated(
+            viewer,
+            market_id,
+            ask_page_size,
+            bid_page_size,
```

Comment:
If I pass `max_asks = 50, max_bids = 150, page_size = 100`, the function will return 100 asks and 200 bids, correct?

If so, that is a violation of user intent and should be fixed

I believe the open orders with pagination function handles this situation properly and you can probably use similar logic here

Reply:
It's like this because we don't know how many orders are covered by the returned price levels result. So this will indeed sometimes process more orders than desired but I figured as long as we stay within the page size and return  an equal or greater number of processed orders then it's fine.

We don't have a way to do this exactly I don't think.

Reply:
Fixed